basket/news/backends/ctms.py

```diff
@@ -682,15 +682,22 @@ def get(
                     )
                 )
 
-            params = {id_name: id_value}
-            contacts = self.interface.get_by_alternate_id(**params)
-
-            if not contacts:
-                contact = None
-            elif len(contacts) == 1:
-                contact = contacts[0]
-            else:
-                raise CTMSMultipleContactsError(id_name, id_value, contacts)
+            # Try alternate IDs in order
+            first_run = True
+            first_contacts = None
+            contact = None
+            for params in alt_ids:
+                contacts = self.interface.get_by_alternate_id(**params)
+                if first_run:
```

Comment:
why would it only be the first one that could return multiples?

Reply:
I'm not sure there is a best return if multiple IDs checks fail, because each returned no contacts or multiple contacts. So, it returns the result of the first one - a ``CTMSMultipleContactsError`` if the first one returned multiple, or ``None`` if it returned no contact list. This matches the previous behavior, when it would check one of the IDs and then give up.